app/moves/middleware.js

```diff
@@ -116,24 +115,52 @@ const movesMiddleware = {
           return moveService
             .getMovesCount({ dateRange, status: moveType.filter, locationId })
             .then(count => {
-              const { label } = moveType
               return {
-                label,
+                ...moveType,
                 value: count,
                 active: moveType.filter === req.params.status,
                 href: `${req.baseUrl}/${period}/${date}${
                   locationId ? '/' + locationId : ''
                 }/${moveType.filter}`,
               }
             })
-            .then(presenters.moveTypesToFilterComponent)
         })
       )
       next()
     } catch (error) {
       next(error)
     }
   },
+  setTotalMoves: (req, res, next) => {
+    if (!get(res, 'locals.moveTypeNavigation')) {
+      return next()
+    }
+    const totalMoves = {
+      label: 'moves::dashboard.filter.total',
+      filter: 'total',
+      href: find(res.locals.moveTypeNavigation, { filter: 'proposed' }).href,
+    }
+    totalMoves.value = res.locals.moveTypeNavigation.reduce(
+      (accumulator, moveType) => {
+        return (accumulator += moveType.value)
+      },
+      0
+    )
+    res.locals.moveTypeNavigation = reject(res.locals.moveTypeNavigation, {
+      filter: 'proposed',
+    })
+    res.locals.moveTypeNavigation.unshift(totalMoves)
+    next()
+  },
+  setTranslatedMoveTypes(req, res, next) {
+    if (!get(res, 'locals.moveTypeNavigation')) {
+      return next()
+    }
+    res.locals.moveTypeNavigation = res.locals.moveTypeNavigation.map(
```

Comment:
You could also move this to the controller if you wanted to move that logic there?

Reply:
This is a less clear cut case. `setTranslatedMoveTypes` is actually shared across 2 routes, and I can see it being reused in similar future views.